### PR TITLE
Focused launch: fix wrong number of domain items while loading

### LINKED ISSUE #47998: Focused Launch: Domain step in Summary view estimates wrong number of domain items while loading
## What

- Open Focused Launch modal
- Change Site Title to something else and move focus away from site title input field
- Domain step starts updating the suggestions: while this happens, the number of items in the domain steps jumps from 3 to 4, then back to 3

DomainStepItemsJump

## Why

There must be some minor bug in how the expected number of domain items is computed

## Acceptance Criteria

- [ ] The expected number of domain items should not jump back and forth while interacting with the Summary view

## Context

[These lines of code](https://github.com/Automattic/wp-calypso/blob/trunk/packages/domain-picker/src/domain-picker/index.tsx#L237-L239) may be a good place to start

### PR DESCRIPTION
**Pull request by razvanpapadopol**
**Note for ET release**: N/A (this PR is fixing a bug that hasn't been released yet)

#### Changes proposed in this Pull Request

* Domain picker should call `setPersistentSelectedDomain` only after `domainSuggestions` are loaded.

#### Testing instructions

* Open Focused Launch 
* Watch how domain suggestions are loading in both Domain step in Summary view and Domain Details view
* The placeholder items should match the rendered items as described in #47998

Fixes #47998

### CODE REVIEW

**Review by ciampo — COMMENTED** [review]
Tested correctly, changes LGTM — will wait for @tjcafferkey to review and give the final approval.

**Comment by tjcafferkey** [comment]
The code changes look good to me, and this does fix the scenario addressed in the issue but this issue does still exist in another scenario.

1. Load Focused Launch
2. Click "View All Domains"
3. Select domain at the bottom of the list
4. Reload the page
5. Open Focused Launch again
6. 3 domain placeholders are present, but 4 domains get rendered.

**Review by tjcafferkey — APPROVED** [review]

**Comment by razvanpapadopol** [comment]
> 3 domain placeholders are present, but 4 domains get rendered.

This is expected since until domain suggestions are loaded, we don't know if the selected domain is included or not in the 3 options to be rendered. This is also applicable for Plan step, where we [append the selected plan (if not popular) when cart products are loaded](https://github.com/Automattic/wp-calypso/pull/47990).

**Comment by tjcafferkey** [comment]
>This is expected since until domain suggestions are loaded, we don't know if the selected domain is included or not in the 3 options to be rendered.

Cool, again I think it's an edge case and I don't believe it to be a huge issue regardless just wanted to make sure we were aware of it. 🚢

**Pull request merged**
